### PR TITLE
Address 15 documentation, help wanted, and question issues

### LINKED ISSUE #32: How to install the this on my Mac ?
Hello Team ,

came across this REPO and wanted to explore more but i wanted to know the installation steps to try this out on my mac.

can anyone please help me with the steps for the same?

Thanks in advance :)

### LINKED ISSUE #33: Please remove the youtube videos from the readme
They are completely useless videos made from youtubers and would made a quant laugh. 
First of all they start by applying the "super easily AI made trading system" to an already rockstar instrument (ie Bitcoin, AAPL, TSLA, whatever), and they intentionally leave the "Buy and Hold" stats out. 
If they'd included it you'd see that you wouldn't need ANY system AT ALL to make insane returns with those instruments if you bought them at the beginning.. so they're cheating from the start.

Plus you cannot use Tradingview (IN SAMPLE DATA) to backtest/create a proper trading system. 
Proper/robust trading systems need to be build with a "Walk Forward" approach using OUT OF SAMPLE data, carefully crafted so that it cannot peak into the future. You'd need specialized software for that.

http://www.codefortraders.com/Walk-Forward_Analysis/WFA_Introduction.htm

Could AI help here? Definitely. But not like this.

### LINKED ISSUE #34: FinGPT-v1 , errors with download_titles.py
I don't understand what does `al_re` mean ?

```python
# ATTENTION! Should replace this with your results path!
al_re = os.listdir(r"D:\python_project\FinRL-Meta\experiment\scrape\results")
al_re = [al.split(".")[0] for al in al_re]
```

### LINKED ISSUE #35: where is Eastmoney_Streaming
where is [code](https://github.com/AI4Finance-Foundation/FinGPT/blob/master/fingpt/FinGPT-v1/data_preparations/download_titles.py) Eastmoney_Streaming

### LINKED ISSUE #47: 在FinGPT-v1的例子，执行 finetune.sh脚本训练时间
我目前用两只股票(茅台和福耀玻璃)，大约有70M，在执行 finetune.sh脚本的时候，一切都正常，但是训练时间很短，只有几分钟，这个正常吗，生成的outdir 看着没啥问题，运行环境是 Tesla V100 SXM2 32GB 使用Fn16

### LINKED ISSUE #48: 在执行infer.ipynb 时，文件里面的两个目录问题
root_path = "/root/autodl-tmp/results"
instructions = pd.read_csv(f"{root_path}/results_new.csv")
instructions.shape
ori_instructions = pd.read_csv(f"{root_path}/results.csv")
ori_instructions.shape

这两个文件从哪里获取的，没有看到生成的代码

### LINKED ISSUE #56: on_filled
why it doesn't have the same signature as on_execution

### LINKED ISSUE #60: 希望取得联系
尊敬的FinGPT 应用开发者，我是 InternLM 社区开发者&志愿者尖米, 大佬开源的工作对我的启发很大，希望可以探讨使用 InternLM 实现FinGPT 的可能性和实现路径，我的微信是mzm312，希望可以取得联系进行更深度的交流。

### LINKED ISSUE #62: training resources
Hello, I would like to ask can the full training of fingpt be done on colab with A100? How long will it take.

### LINKED ISSUE #134: quetions about FinGPT-RAG
Hello, I have some questions about FinGPT-RAG.
You mentioned in your paper that you have fine-tuned the LlaMa-7B via instuction datasets and got good results compared to other LLMs, and also mentioned that adding RAG module got good results.
But what I've seen so far is that you seem to be just open sourcing the code framework, not the fine-tuned FinGPT with RAG that can be used directly.
Will you open source the fine-tuned FinGPT with RAG later?
And if I don't have enough computing resources to perform instruction fine-tuning on Llama-7B, can I only use RAG method to enhance the LLM's ability in financial sentiment analysis?

### LINKED ISSUE #142: It's hard to download eastmoney content
When we download title of eastmoiney according to "fingpt/FinGPT_Sentiment_Analysis_v1/FinGPT_v1.0" with kuaidaili, we got the hs300    freely.
But When we download the content og eastmony  according to "fingpt/FinGPT_Sentiment_Analysis_v1/FinGPT_v1.0" , we got nothing. And there is no error reported.
What can we do now if we want fine-tuning FinGPT according to FinGPT_v1.0?

### LINKED ISSUE #146: Reproducing sentiment finetuning train_lora extremely slow
I am trying to reproduce the finetuning for the fingpt-sentiment_llama2-13b_lora

The table claims we can do this in just a single RTX 3090 within a day.
I am using a L4 GPU instead.

I downloaded the models to base_models and the dataset to data correctly

I used the script like this

```
deepspeed -i train_lora.py \
--run_name sentiment-llama2-13b-20epoch-64batch \
--base_model llama2-13b-nr \
--dataset sentiment-train \
--max_length 512 \
--batch_size 16 \
--learning_rate 1e-4 \
--num_epochs 20 \
```

I got an OOM.

So i set the `load_in_8_bit=True`

But I am getting extremely slow fine tuning speed A single epoch is estimated to take 2 days.

### LINKED ISSUE #151: Multi-Tasking with FinGPT
Can I use FinGPT to extract financial information such as metrics, values, company name, etc. from financial reports? If yes, how. I am finding it difficult to run FinGPT on colab for my customized prompt.

I want to check for consistencies of the financial information discussed in the reports.

Many thanks!

### LINKED ISSUE #153: Do you support China's stock forecast

### LINKED ISSUE #156: Dataset for FinForcaster.
**Is your feature request related to a problem? Please describe.**
It would be great to be able to use the data to finetune a Mistral model because Mistral models have better performance than llama 7b, and I can also modify the dataset with newer data.

**Describe the solution you'd like**
Kindly upload the dataset that was used to train FinForcaster.

**Describe alternatives you've considered**
I did check FinGpt page on huggingface but I didn't find it there.
A link to the original DOW dataset used would be good enough.
Thank you.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by jellyfishing2346**
This PR addresses all open issues with labels "documentation", "help wanted", or "question" by improving documentation, adding clarifications, and providing comprehensive guides.

### Documentation Improvements (#33)
- Removed YouTube video links from README.md as they were deemed not useful for serious quantitative finance applications
- Cleaned up ChatGPT trading bot section and other YouTube references

### macOS Installation Support (#32)
- Added comprehensive macOS-specific installation instructions in SETUP.md
- Included Apple Silicon (M1/M2/M3) support with MPS (Metal Performance Shaders) guidance
- Added macOS troubleshooting section for common installation issues
- Provided device configuration examples for Mac users

### Code Clarification (#34)
- Added detailed comments explaining the  variable in download_titles.py
- Clarified that it contains already-processed stock codes to prevent re-downloading
- Added documentation for result_path variable usage

### Inference Documentation (#48)
- Added comprehensive documentation in infer.ipynb explaining where CSV files come from
- Clarified that results.csv and results_new.csv are generated during training
- Provided guidance on what to do if files are missing

### Training Time Documentation (#47)
- Added expected training times section to FinGPT v1.0 README
- Included reference times for different hardware configurations
- Added troubleshooting guidance for unusually short training times
- Listed factors that affect training duration

### Comprehensive FAQ (#151, #134, #62, #56)
- Created new FAQ.md file addressing common questions
- Covered multi-tasking capabilities, RAG usage, hardware requirements
- Addressed China stock market support questions
- Included technical questions about function signatures
- Added training resource requirements for Colab/A100
- Referenced FAQ from main README.md

### Eastmoney_Streaming Documentation (#35)
- Added clarification about Eastmoney_Streaming location in FinNLP library
- Updated download_titles.py with better comments about FinNLP dependency
- Added instructions for proper FinNLP repository setup

### Eastmoney Download Troubleshooting (#142)
- Added comprehensive troubleshooting section for eastmoney download issues
- Covered proxy configuration, rate limiting, network issues
- Provided alternative data source suggestions
- Added script configuration guidance

### Training Performance Documentation (#146)
- Added detailed performance troubleshooting section in SETUP.md
- Included GPU configuration checks and optimization tips
- Documented quantization trade-offs (8-bit vs 4-bit)
- Provided batch size and DeepSpeed configuration guidance
- Added expected performance benchmarks for reference

### Dataset Availability Documentation (#156)
- Enhanced FAQ with detailed dataset availability information
- Clarified FinGPT-Forecaster dataset location and contents
- Provided guidance on using datasets for other models (Mistral, etc.)
- Explained dataset structure and extensibility

### Market Support Documentation (#153)
- Added market support clarification to README.md
- Documented that FinGPT-Forecaster is trained on US market (DOW 30)
- Provided guidance for adapting to other markets including China
- Referenced FinGPT v1.x for Chinese market applications

### Contact and Collaboration Guidelines (#60)
- Added comprehensive contact and collaboration section to CONTRIBUTING.md
- Provided recommended channels for different types of inquiries
- Added guidance for personal contact requests
- Commented on issue #60 with appropriate contact channel suggestions

## Files Modified:
- README.md: Removed YouTube links, added FAQ reference, market support info
- SETUP.md: Added macOS installation guide, performance troubleshooting
- FAQ.md: New comprehensive FAQ document
- CONTRIBUTING.md: Added contact and collaboration guidelines
- fingpt/FinGPT_Sentiment_Analysis_v1/FinGPT_v1.0/README.md: Training times, eastmoney troubleshooting, FinNLP documentation
- fingpt/FinGPT_Sentiment_Analysis_v1/FinGPT_v1.0/data_preparations/download_titles.py: Code clarification comments
- fingpt/FinGPT_Sentiment_Analysis_v1/FinGPT_v1.0/inferencing/infer.ipynb: File source documentation

## Issues Addressed:
Closes #33 - Remove YouTube videos from README
Closes #32 - How to install on Mac
Closes #34 - FinGPT-v1 errors with download_titles.py (al_re variable)
Closes #48 - Missing files in infer.ipynb
Closes #47 - Training time questions
Closes #151 - Multi-tasking with FinGPT
Closes #134 - Questions about FinGPT-RAG
Closes #62 - Training resources (Colab/A100)
Closes #56 - on_filled function signature
Closes #35 - Eastmoney_Streaming location
Closes #142 - Eastmoney download issues
Closes #146 - Slow training performance
Closes #156 - Dataset for FinForecaster
Closes #153 - China stock forecast support
Closes #60 - Contact request (InternLM collaboration)

### CODE REVIEW

**Comment by jellyfishing2346** [comment]
@BruceYanghy feel free to leave feedback on this PR.
